kernel/src/process/signal/poll.rs

```diff
@@ -64,69 +148,28 @@ impl Pollee {
         poller.pollees.push(Arc::downgrade(&self.inner));
     }
 
-    /// Register an IoEvents observer.
-    ///
-    /// A registered observer will get notified (through its `on_events` method)
-    /// every time new events specified by the `mask` argument happen on the
-    /// pollee (through the `add_events` method).
+    /// Notifies pollers of some events.
     ///
-    /// If the given observer has already been registered, then its registered
-    /// event mask will be updated.
+    /// This method wakes up all registered pollers that are interested in the events.
     ///
-    /// Note that the observer will always get notified of the events in
-    /// `IoEvents::ALWAYS_POLL` regardless of the value of `mask`.
-    pub fn register_observer(&self, observer: Weak<dyn Observer<IoEvents>>, mask: IoEvents) {
-        let mask = mask | IoEvents::ALWAYS_POLL;
-        self.inner.subject.register_observer(observer, mask);
-    }
+    /// This method should be called whenever new events arrive.
+    pub fn notify(&self, events: IoEvents) {
```

Comment:
Done:
```diff
@@ -152,7 +152,8 @@ impl Pollee {
     ///
     /// This method wakes up all registered pollers that are interested in the events.
     ///
-    /// This method should be called whenever new events arrive.
+    /// This method should be called whenever new events arrive. The events can be spurious. This
+    /// way, the caller can avoid expensive calculations and simply add all possible ones.
     pub fn notify(&self, events: IoEvents) {
         self.invalidate();
 
@@ -161,7 +162,9 @@ impl Pollee {
 
     /// Invalidates old events.
     ///
-    /// This method should be called whenever old events disappear but no new events arrive.
+    /// This method should be called whenever old events disappear but no new events arrive. The
+    /// invalidation can be spurious, so the caller can avoid complex calculations and simply
+    /// invalidate even if no events disappear.
     pub fn invalidate(&self) {
         // The memory order must be `Release`, so that the reader is guaranteed to see the changes
         // that trigger the invalidation.
```